src/Orleans/LogConsistency/IProtocolServices.cs

```diff
@@ -0,0 +1,141 @@
+        /// </summary>
+        GrainReference GrainReference { get;  }
+
+        
+        /// <summary>
+        /// The multicluster registration strategy for this grain.
+        /// </summary>
+        IMultiClusterRegistrationStrategy RegistrationStrategy { get; }
+
+
+        /// <summary>
+        /// Whether this cluster is running in a multi-cluster network.
+        /// </summary>
+        /// <returns></returns>
+        bool MultiClusterEnabled { get; }
+
+
+        /// <summary>
+        /// The id of this cluster. Returns "I" if no multi-cluster network is present.
+        /// </summary>
+        /// <returns></returns>
+        string MyClusterId { get; }
+
+    
+        /// <summary>
+        /// The current multicluster configuration of this silo 
+        /// (as injected by the administrator) or null if none.
+        /// </summary>
+        MultiClusterConfiguration MultiClusterConfiguration { get; }
+
+        /// <summary>
+        /// List of all clusters that currently appear to have at least one active
+        /// gateway reporting to the multi-cluster network. 
+        /// There are no guarantees that this membership view is complete or consistent.
+        /// If there is no multi-cluster network, returns a list containing the single element "I".
+        /// </summary>
+        /// <returns></returns>
+        IEnumerable<string>  ActiveClusters { get; }
+
+
+        void SubscribeToMultiClusterConfigurationChanges();
+
+        void UnsubscribeFromMultiClusterConfigurationChanges();
+
+
+        #region Logging Functionality
+
+        /// <summary>
+        /// Log an error that occurred in a log-consistency protocol.
+        /// </summary>
+        void ProtocolError(string msg, bool throwexception);
+
+        /// <summary>
+        /// Log an exception that was caught in the log-consistency protocol.
+        /// </summary> 
+        void CaughtException(string where, Exception e);
+
+        /// <summary>
+        /// Log an exception that occurred in user code, for some callback
+        /// </summary>
+        /// <param name="callback">The name of the callback</param>
+        /// <param name="where">The context from which the callback was called</param>
+        /// <param name="e">The caught exception</param>
+        void CaughtUserCodeException(string callback, string where, Exception e);
+
+        /// <summary> Output the specified message at <c>Info</c> log level. </summary>
+        void Info(string format, params object[] args);        
+        /// <summary> Output the specified message at <c>Verbose</c> log level. </summary>
+        void Verbose(string format, params object[] args);
+        /// <summary> Output the specified message at <c>Verbose2</c> log level. </summary>
+        void Verbose2(string format, params object[] args);
+        /// <summary> Output the specified message at <c>Verbose3</c> log level. </summary>
+        void Verbose3(string format, params object[] args);
```

Comment:
Why are we putting logging specific facades here? It will make it even harder than now to move away from our own logging and it is very likely that when we move to a logging abstraction, Verbose2, 3 and customer log levels will vanish and here it would mean breaking an API contract, which makes the move really hard. We should have a better way getting around this.